yes3.py

```diff
@@ -413,7 +413,7 @@ def add_to_bucket_summary(category, bucket_name):
         for rule in lifecycle_rules:
             lifecycle = "no_expiration"
 
-            if rule.get('Expiration').get("Days") or rule.get("NoncurrentVersionExpiration"):
+            if rule.get('Expiration') and rule.get('Expiration').get("Days") or rule.get("NoncurrentVersionExpiration"):
```

Comment:
Good catch here for handling other lifecycle rules, @pilvikala. 

Another user submitted a PR that has been merged in and I recently updated the logic.  I believe that should cover the use cases for what you're trying to fix with this addition of rule.get('Expiration').  Could you confirm the newest code satisfies the use cases you see?  https://github.com/FogSecurity/yes3-scanner/tree/main

Reply:
Resolving this as it's been fixed in https://github.com/FogSecurity/yes3-scanner/tree/main.  I merged main into this PR.